pkg/daemon/ceph/osd/volume.go

```diff
@@ -442,10 +442,15 @@ func UpdateLVMConfig(context *clusterd.Context, onPVC, lvBackedPV bool) error {
 func (a *OsdAgent) initializeDevices(context *clusterd.Context, devices *DeviceOsdMapping) error {
 	storeFlag := "--bluestore"
 
+	logPath := "/tmp/ceph-log"
+	if err := os.MkdirAll(logPath, 0700); err != nil {
+		return errors.Wrapf(err, "failed to create dir %q", logPath)
+	}
+
 	// Use stdbuf to capture the python output buffer such that we can write to the pod log as the logging happens
 	// instead of using the default buffering that will log everything after ceph-volume exits
 	baseCommand := "stdbuf"
-	baseArgs := []string{"-oL", cephVolumeCmd, "lvm", "batch", "--prepare", storeFlag, "--yes"}
+	baseArgs := []string{"-oL", cephVolumeCmd, "--log-path", logPath, "lvm", "batch", "--prepare", storeFlag, "--yes"}
```

Comment:
Does this change the logging that we capture by default in the successful case? Or does it not affect the successful case?

Reply:
We still capture the logging on buf for stdout and print the log file content on error now.